msu/classes/weighted_container.nut

```diff
@@ -35,6 +36,34 @@
 		this.Array.push(_item);
 	}
 
+	function find( _item )
+	{
+		foreach (i, item in this.Array)
+		{
+			if (item[1] == _item) return i;
+		}
+
+		return null;
+	}
+
+	function remove( _idx )
+	{
+		this.Array.remove(_idx);
+	}
+
+	function getProbability( _item )
+	{
+		local idx = this.find(_item);
+		if (idx == null) throw ::MSU.Exception.KeyNotFound;
+		return this.Array[idx][0].tofloat() / this.Total;
+	}
+
+	function sort( _function )
```

Comment:
An alternative implementation of this function could be. However, in this case I am not sure if the default sorting i.e. when _function is null should be by weight (index 0) or by value (index 1).
```
function sort( _function = null )
{
	if (_function != null)
	{
		::MSU.requireFunction(_function);
		this.Array.sort(_function);
	}
	else
	{
		this.Array.sort(@(a, b) a[0] <=> b[0]);
	}
}
```

Reply:
I don't think this needs to implement sort, I think remove is reasonable because I can see a use-case for it, but this shouldn't even expose that it's a sorted array. If you're trying to sort this you're trying to use it for something it wasn't designed for imo